mayastor/src/subsys/config/pool.rs

```
@@ -0,0 +1,181 @@
+                        error.verbose()
+                    );
+                    failures += 1;
+                }
+            }
+        }
+        failures
+    }
+
+    /// Import pools
+    pub fn import_pools(self) {
+        assert_eq!(Cores::current(), Cores::first());
+        Reactor::block_on(async move {
+            let errors = self.create_pools().await;
+            if errors != 0 {
+                warn!(
+                    "Not all pools were imported successfully ({} errors)",
+                    errors
+                );
+            }
+        });
+    }
+
+    /// Capture current pool configuration
+    pub fn get() -> PoolConfig {
+        let pools = PoolsIter::new()
+            .map(|p| {
+                let base = p.get_base_bdev();
+                let name = p.get_name();
+                let replicas = ReplicaIter::new()
+                    .filter(|r| r.get_pool_name() == name)
+                    .map(|r| Replica {
+                        name: r.get_name().to_string(),
+                        share: r.get_share_type(),
+                    })
+                    .collect();
+                Pool {
+                    name: name.to_string(),
+                    disks: vec![base.bdev_uri().unwrap_or_else(|| base.name())],
+                    replicas: as_option(replicas),
+                }
+            })
+            .collect();
+
+        PoolConfig {
+            pools: Some(pools),
+        }
+    }
+}
+
+#[derive(Debug, Default, PartialEq, Serialize, Deserialize, Clone)]
+/// Pools that we create. Future work will include the ability to create RAID0
+/// or RAID5.
+pub struct Pool {
+    /// name of the pool to be created or imported
+    pub name: String,
+    /// bdevs to create outside of the nexus control
+    pub disks: Vec<String>,
+    /// list of replicas (not required, informational only)
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub replicas: Option<Vec<Replica>>,
```

I don't think we need the replicas, they're shared via the properties?

---

As the comment says, the list of replicas is informational only - it is not used on start up as the pool knows what replicas have been created. This is potentially misleading, so for now the pool config is updated everytime a replica changes in order to maintain consistency. I would like to remove all references to replicas entirely, but I really want to do that as a separate PR in order to make it easier to reinstate should we wish to at some point in the future.

---

:+1:

---

Remove what is not needed, please. Just the pool information is what we want.

---

No replica ingormation is generated. However if replica information is present in the yaml, rather than fail with invalid config, the replica information will be parsed but otherwise ignored.